### ISSUE TITLE
Method survey 2016

### ISSUE THREAD

**Issue by agladysh** [NaNoGenMo/2016#154]
List of generation methods used in NaNoGenMo 2016.

This top post shall serve as the canonical resource. Please comment below for inclusion. 

* Markov chains:
* Text templates: #136 #137
* Recursion:
* Literary text post-processing:
* Non-literary text or data post-processing:
* Neural networks:
* Simulation:

Any single project may be listed under several categories.

**Comment by agladysh** [comment]
The list of categories is tentative, if you feel that there is no good fit for your project, please list it under a new category in the comments. Contributions of brief definition for any category are also welcome.

**Comment by dranorter** [comment]
I, #128, was recursive text templates.

**Comment by ikarth** [comment]
#6 is text templates plus simulation

**Comment by enkiv2** [comment]
Literary text post-processing: #139 #135 #12
Non-literary text post-processing: #49 
Text templates: #113 #12 #8

**Comment by serin-delaunay** [comment]
Text templates (with a small amount of recursion): #125
Markov chain: #143

**Comment by edgriebel** [comment]
#131 Markov Chains

**Comment by binaricorn** [comment]
[#155](https://github.com/NaNoGenMo/2016/issues/155) text templates and simulation

**Comment by agladysh** [comment]
@dranorter, @serin-delaunay Do you think that the "recursive" part in "recursive text template" is important?

NB: That's a serious question, not trolling. I am not sure yet about how to approach the classification.

**Comment by agladysh** [comment]
I updated the list. Many thanks for contributions! Please feel free to comment on the format — and to classify more entries! :)

**Comment by moonmilk** [comment]
#149: non-literary text (tweets), post-processing

**Comment by agladysh** [comment]
Aha! About the recursion: I just noted that there is a separate "recursion" method. I guess we do need to work out definitions here. 

In https://habrahabr.ru/post/313862/ "recursion" section actually describes two separate approaches. 

First is where the text consists of many shorter stories, with the end of the previous story naturally leads to the beginning of the next one. #136 uses this approach. I propose to call this method iterative text templates, as there is no "depth" in the recursion, it is used more as a literary device than as a method of text generation.

The second approach is true recursion, where the content is of a fractal nature, so to say. See  [Transorbital anaphase provine biforn the pure-bred synostosis](https://github.com/samcoppini/Definition-book/blob/master/The%20transorbital%20anaphase%20provine%20biforn%20the%20pure-bred%20synostosis.md) as an example, which is basically a set of nested dictionary definitions, or [Redwreath and Goldstar Have Traveled to Deathsgate](http://eblong.com/zarf/essays/r-and-g.html), where two characters have a discussion about their discussion, ending up with «You want to know whether I am asking whether you are asking whether you shall tell me whether you want to know whether I believe I can answer that?»

**Comment by dranorter** [comment]
Right; my #128 is making a point of having several types of "true recursion;" sub-quests, locations within locations, and stories within stories.

> [...] or Redwreath and Goldstar Have Traveled to Deathsgate, where two characters have a discussion about their discussion, ending up with «You want to know whether I am asking whether you are asking whether you shall tell me whether you want to know whether I believe I can answer that?»

Nice! I wish I had thought to include that!

**Comment by BenKybartas** [comment]
#133 and #147 used text templates and recursion.

**Comment by serin-delaunay** [comment]
I didn't mean to distinguish recursive and non-recursive text templates as methods; recursion was one of the methods listed, and I found it unclear what it meant as a method. Since a few of the templates used in #125 were recursive (primarily object lists, usernames, and passwords), and since fortunes are told for children, grandchildren, and great-grandchildren using recursion, I thought it might be appropriate to list #125 under recursion as well as under text templates.

**Comment by dranorter** [comment]
It would look cleaner to just list the recursive template examples under "recursion". Though I'd like to know what other sorts of recursion might be used. A train, generate, train, generate, train loop? Repeated OCR of printouts of Gutenberg texts?

**Comment by nagolinc** [comment]
#130  "Text templates with recursion and simulation"

I have a tree of possible events in the story and then a recursive template that says which events
to choose when creating the story.  Recursion because the story follows a "goal>subquest>subsubquest" model and simulation because different events in the story affect the "world state" which can be accessed later on to generate text.

**Comment by spikelynch** [comment]
#114 used neural networks, simulation and text templates.

**Comment by joeld42** [comment]
I used a few of these ( #45 ). "Filler" scenes were generated with Tracery or by filling in simple templates. Combat scenes were done with a simulation. Markov Chains (seeded with a large list of cities) to generate names for people and places. Didn't reference other works, unless you count the covers, which mixed and matched a few templates and PD or creative commons artwork.

**Comment by dluman** [comment]
[https://github.com/NaNoGenMo/2016/issues/24](#24) uses sentence similarity derived from WordNet and Corpus statistics in addition to aleatory page selection from a given set of documents; so it's two different uses/methods approaching corpora—there's probably a way better way to describe this.

**Comment by tra38** [comment]
#15 uses a "Story Compiler" approach, which I suppose, for lack of a better category, "literary-text post-processing".

**Comment by agladysh** [comment]
Many thanks for the submissions, feedback and your work on the NaNoGenMo, all! I updated the issue text. Any further comments, critique and other feedback are appreciated — not to mention more submissions to this list!

**Comment by hugovk** [comment]
* https://github.com/NaNoGenMo/2016/issues/61 is "non-literary text or data post-processing (time)"
* https://github.com/NaNoGenMo/2016/issues/138 is "non-literary text or data post-processing (tweets)"

**Comment by agladysh** [comment]
Updated, thank you! Any more submissions? :)
